bootstrap.sh

```diff
@@ -1,23 +1,43 @@
-#! /bin/sh
+#! /bin/bash
 
-set -eu
+# Wait until after we finish testing a possibly empty variable to do `set -u`
```

Comment:
I'm not sure whether I understand this change. What problem does this solve?

Reply:
The `-u` flag causes errors whenever the script encounters an undefined variable. This defeats the point of testing to see if certain variables are empty or not. For example the test on line 11 of:
```bash
if [ "x${gnulibtool}" = "x" ]; then
```
...will cause the entire script to error out if the script is run with `set -u`, like this:
```
$ sh ./bootstrap.sh
./bootstrap.sh: line 11: gnulibtool: unbound variable
```